repos/system_upgrade/el7toel8/actors/storagescanner/libraries/library.py

```diff
@@ -79,22 +80,32 @@ def _get_fstab_info(fstab_path):
     ''' Collect storage info from /etc/fstab file '''
     if _is_file_readable(fstab_path):
         with open(fstab_path, 'r') as fstab:
-            for entry in fstab:
+            keys = ['fs_spec', 'fs_file', 'fs_vfstype', 'fs_mntops', 'fs_freq', 'fs_passno']
+            for line, entry in enumerate(fstab, 1):
                 if entry.startswith('#'):
                     continue
 
+                if '#' in entry:
+                    # Ignore comments at end of lines.
+                    entry = entry.split('#')[0]
+
                 entry = entry.strip()
                 if not entry:
                     continue
 
-                fs_spec, fs_file, fs_vfstype, fs_mntops, fs_freq, fs_passno = entry.split()
-                yield FstabEntry(
-                    fs_spec=fs_spec,
-                    fs_file=fs_file,
-                    fs_vfstype=fs_vfstype,
-                    fs_mntops=fs_mntops,
-                    fs_freq=fs_freq,
-                    fs_passno=fs_passno)
+                entries = entry.split()
+
+                if len(entries) not in [4, 5, 6]:
+                    summary = "Please fix your /etc/fstab file on the line: {}, it must have" \
+                        "at least 4 values and at most 6 this line has {}".format(line, len(entries))
+                    reporting.report_generic(
+                        title='Invalid Fstab', summary=summary, severity='high', flags=['inhibitor']
+                    )
```

Comment:
You should use remediation report. The summary you wrote is actually more like remediation and it is not telling what is wrong. Summary should be about the error that occur (received X fields, expected...) and remediation instruction what user should do. So, in example I would point users to `man 5 fstab` and explain we expect format documented in the man page with comments on separate lines (ideally).